lib/RuntimePlugin.js

```diff
@@ -218,7 +218,17 @@ class RuntimePlugin {
 			compilation.hooks.runtimeRequirementInTree
 				.for(RuntimeGlobals.systemContext)
 				.tap("RuntimePlugin", chunk => {
-					if (compilation.outputOptions.library.type === "system") {
+					const { outputOptions } = compilation;
+					const { library: globalLibrary } = outputOptions;
+					const entryOptions = chunk.getEntryOptions();
+					const libraryType =
+						entryOptions &&
+						entryOptions.library &&
+						entryOptions.library.type !== undefined
+							? entryOptions.library.type
+							: globalLibrary.type;
```

Comment:
```suggestion
					const libraryType =
						entryOptions &&
						entryOptions.library !== undefined
							? entryOptions.library.type
							: globalLibrary.type;
```

When a `library` is specified for an entry it overrides the global one, and doesn't merge with it.